### ISSUE TITLE
跑deepctr样例，auc只有0.5， 提示no gradient found

### ISSUE THREAD

**Issue by simon1024** [alibaba/x-deeplearning#183]
对deepctr样例，做了微小的改动，打印出auc，结果发现，auc一直在0.5上不去，同时还会提示全连接层的参数没有梯度，请问这是咋回事？
输出信息：
<pre>
[WARNING]: no gradient found for var:dense/kernel under scope: , maybe not used?
[WARNING]: no gradient found for var:dense/bias under scope: , maybe not used?
[WARNING]: no gradient found for var:dense_1/kernel under scope: , maybe not used?
[WARNING]: no gradient found for var:dense_1/bias under scope: , maybe not used?
[WARNING]: no gradient found for var:dense_2/kernel under scope: , maybe not used?
[WARNING]: no gradient found for var:dense_2/bias under scope: , maybe not used?
loss:0.6931536,auc:0.50344515
loss:0.69737524,auc:0.49663246
loss:0.8758856,auc:0.50528824
……
</pre>
改过后代码如下：
<pre>
def train():
    batch = reader.read()
    sess = xdl.TrainSession()
    emb1 = xdl.embedding('emb1', batch['sparse0'], xdl.TruncatedNormal(stddev=0.001), 8, 1024, vtype='hash')
    emb2 = xdl.embedding('emb2', batch['sparse1'], xdl.TruncatedNormal(stddev=0.001), 8, 1024, vtype='hash')
    loss, auc, y = model(batch['deep0'], [emb1, emb2], batch['label'])
    train_op = xdl.SGD(0.5).optimize()
    log_hook = xdl.LoggerHook([loss,auc], "loss:{0},auc:{1}", 1)
    sess = xdl.TrainSession(hooks=[log_hook])
    while not sess.should_stop():
        sess.run(train_op)

@xdl.tf_wrapper()
def model(deep, embeddings, labels):
    input = tf.concat([deep] + embeddings, 1)
    fc1 = tf.layers.dense(
        input, 128, kernel_initializer=tf.truncated_normal_initializer(
            stddev=0.001, dtype=tf.float32))
    fc2 = tf.layers.dense(
        fc1, 64, kernel_initializer=tf.truncated_normal_initializer(
            stddev=0.001, dtype=tf.float32))
    fc3 = tf.layers.dense(
        fc2, 32, kernel_initializer=tf.truncated_normal_initializer(
            stddev=0.001, dtype=tf.float32))
    y = tf.layers.dense(
        input, 1, kernel_initializer=tf.truncated_normal_initializer(
            stddev=0.001, dtype=tf.float32))
    loss = tf.losses.sigmoid_cross_entropy(labels, y)

    y_pred = tf.sigmoid(y)
    _, auc = tf.metrics.auc(labels=labels, predictions=y_pred)

    return loss, auc, y

train()
</pre>

**Comment by songyue1104** [comment]
模型写错了
y = tf.layers.dense(
        input, 1, kernel_initializer=tf.truncated_normal_initializer(
            stddev=0.001, dtype=tf.float32))

另外训练数据是完全随机生成的，请使用真实数据验证

**Comment by simon1024** [comment]
> 模型写错了
> y = tf.layers.dense(
> input, 1, kernel_initializer=tf.truncated_normal_initializer(
> stddev=0.001, dtype=tf.float32))
> 
> 另外训练数据是完全随机生成的，请使用真实数据验证



> 模型写错了
> y = tf.layers.dense(
> input, 1, kernel_initializer=tf.truncated_normal_initializer(
> stddev=0.001, dtype=tf.float32))
> 
> 另外训练数据是完全随机生成的，请使用真实数据验证

不好意思，上面的代码是我调试的时候，忘记改回来了。问题181能否帮忙看看：https://github.com/alibaba/x-deeplearning/issues/181

**Comment by simon1024** [comment]
> 模型写错了
> y = tf.layers.dense(
> input, 1, kernel_initializer=tf.truncated_normal_initializer(
> stddev=0.001, dtype=tf.float32))
> 
> 另外deepctr这个例子主要是说明api用法，训练数据是完全随机生成的，不保证收敛效果

我又想了下，即使是随机生成的数据，也不该每次auc都是0.5吧，我自己构造了个样本数据，里面有2条样本，其中1条是负样本，另1条是正样本，连续训练100个epochs，batch：2， 按理肯定会过拟合的，可是，auc依然是0.5。
样本数据：
<pre>
skey1|gkey1|sparse0@759:0.762162,548:0.101924,716:0.194687,421:0.264081,261:0.404047,385:0.499228,449:0.959710,324:0.599362,731:0.446862;sparse1@546:0.951531,432:0.898802,\
105:0.783065,857:0.471639|deep0@0.662782,0.342298,0.468571,0.718452,0.713778,0.163456,0.887874,0.960262,0.044802,0.853923,0.400626,0.711758,0.221172,0.688655,0.571846,0.91\
1907,0.302197,0.677726,0.610022,0.452376,0.974598,0.304923,0.926549,0.402046,0.137225,0.083297,0.956003,0.084375,0.474048,0.183849,0.023845,0.262389,0.094568,0.385041,0.60\
4758,0.807893,0.546411,0.263949,0.620935,0.662830,0.221174,0.947233,0.609922,0.458325,0.163481,0.514653,0.258731,0.243340,0.949402,0.715515,0.525947,0.075673,0.805997,0.89\
9717,0.344329,0.145784,0.130369,0.234209,0.653809,0.297923,0.356226,0.412527,0.986982,0.645392,0.447851,0.945122,0.698532,0.270804,0.824928,0.355763,0.949162,0.339264,0.03\
5675,0.889754,0.001660,0.881021,0.087891,0.798518,0.472960,0.926075,0.475933,0.680346,0.125074,0.389097,0.480265,0.447326,0.554875,0.913568,0.213271,0.072741,0.928426,0.43\
3917,0.437505,0.196376,0.288154,0.271000,0.271465,0.677179,0.148842,0.208836,0.258478,0.554515,0.271461,0.493273,0.263461,0.014935,0.378840,0.726748,0.601786,0.327549,0.36\
7978,0.210795,0.902559,0.990410,0.990561,0.203781,0.976384,0.529055,0.439665,0.979978,0.919847,0.628117,0.531591,0.559527,0.854383,0.567795,0.669983,0.131906,0.810021,0.45\
7204,0.362367,0.241847,0.259656,0.095626,0.066626,0.639766,0.264775,0.555648,0.450152,0.721681,0.886644,0.001726,0.037867,0.024443,0.223571,0.082423,0.138201,0.767822,0.55\
4831,0.122466,0.142976,0.508260,0.504362,0.967684,0.447289,0.653109,0.651150,0.712938,0.553411,0.832021,0.061552,0.590981,0.704351,0.847320,0.522703,0.585140,0.965691,0.96\
2297,0.600640,0.508416,0.461186,0.252704,0.672691,0.631777,0.460096,0.574279,0.726332,0.429794,0.212263,0.237247,0.759556,0.078822,0.932726,0.200260,0.420823,0.578012,0.61\
4804,0.543419,0.448529,0.292871,0.238784,0.565181,0.316637,0.884232,0.672599,0.583305,0.608038,0.217691,0.100539,0.430582,0.476008,0.724872,0.480458,0.252887,0.330800,0.81\
4417,0.081829,0.570468,0.336623,0.598733,0.356824,0.500899,0.294857,0.760062,0.451524,0.641034,0.087247,0.289112,0.864109,0.854319,0.093764,0.378239,0.075818,0.455410,0.10\
4947,0.316122,0.462723,0.420934,0.647593,0.916033,0.684150,0.219117,0.449281,0.940042,0.845376,0.572261,0.072145,0.046971,0.692016,0.404238,0.668091,0.983969,0.686270,0.16\
3032,0.064395,0.463078,0.170263,0.834831,0.521397,0.390686,0.855010,0.123449,0.766170,0.223663,0.076849,0.653404|0|1544094735
skey2|gkey2|sparse0@856:0.352409,953:0.663148,523:0.722702,99:0.187631,753:0.935152,746:0.576277,458:0.820544,962:0.843274,325:0.527312,163:0.092034,620:0.242616,984:0.1610
33,166:0.154479,842:0.875578;sparse1@515:0.208188,1019:0.591209,145:0.156365,605:0.997317,1021:0.940796,729:0.273456,1005:0.047435,433:0.377895,1003:0.431236,500:0.032101,5
56:0.318865,145:0.522828,221:0.465735,685:0.818602,77:0.669590|deep0@0.999240,0.440470,0.047516,0.430420,0.634393,0.204206,0.665262,0.337837,0.169835,0.089634,0.878299,0.11
1705,0.523178,0.397379,0.832842,0.916505,0.390858,0.765615,0.332115,0.320150,0.951650,0.246418,0.512040,0.749791,0.673629,0.535592,0.865059,0.583392,0.691534,0.559879,0.246
394,0.822141,0.191856,0.138813,0.977759,0.104353,0.643541,0.891259,0.387623,0.138292,0.287816,0.773174,0.577388,0.399102,0.661399,0.436638,0.247339,0.000672,0.343219,0.2713
45,0.483667,0.311244,0.595922,0.712664,0.177472,0.232139,0.686599,0.857067,0.745553,0.588793,0.775999,0.510141,0.002143,0.174302,0.393704,0.840412,0.946845,0.016176,0.84112
9,0.698696,0.022466,0.431059,0.038136,0.526626,0.002277,0.137417,0.853434,0.192572,0.495922,0.832388,0.056398,0.564264,0.424322,0.520221,0.358680,0.811522,0.269201,0.204860
,0.697190,0.815236,0.587757,0.628998,0.290156,0.128817,0.977050,0.951258,0.501323,0.977213,0.705750,0.926940,0.904108,0.891195,0.229011,0.194533,0.544853,0.837290,0.627977,
0.806972,0.838957,0.867092,0.192085,0.943976,0.931445,0.915464,0.136756,0.966073,0.165868,0.208994,0.303334,0.992627,0.080906,0.722388,0.315930,0.413592,0.613303,0.562165,0
.844661,0.125174,0.963790,0.434821,0.734555,0.375568,0.309204,0.406794,0.928774,0.183238,0.905576,0.656561,0.300933,0.189950,0.558191,0.842550,0.747929,0.393342,0.866781,0.
298084,0.114384,0.530025,0.934307,0.026605,0.224099,0.074621,0.373608,0.696212,0.384710,0.658822,0.071645,0.514396,0.968043,0.660373,0.380076,0.062846,0.835793,0.160011,0.9
72219,0.552067,0.852591,0.498116,0.467868,0.871608,0.558435,0.296820,0.606116,0.869073,0.521347,0.962195,0.555587,0.718314,0.014050,0.584082,0.540957,0.390047,0.281304,0.02
6981,0.814965,0.136477,0.696427,0.486134,0.173861,0.349167,0.985647,0.426176,0.523973,0.942895,0.862479,0.206533,0.215264,0.095748,0.811125,0.339203,0.185293,0.321666,0.950
192,0.602032,0.185758,0.147267,0.725569,0.275365,0.454471,0.293140,0.850988,0.407612,0.778952,0.788696,0.849566,0.776480,0.259140,0.723711,0.859693,0.268434,0.498420,0.7846
41,0.465112,0.339739,0.813121,0.181796,0.996655,0.796091,0.264137,0.762158,0.731558,0.932281,0.776534,0.151321,0.153504,0.940771,0.205348,0.301515,0.098815,0.396543,0.83577
1,0.861178,0.180162,0.364074,0.570220,0.055760,0.072960,0.578157,0.777676,0.435500,0.865028,0.283787,0.616606,0.019386,0.217615,0.463562|1|1544094735
</pre>
